### DISCUSSION TITLE
SSL - Certificate is invalid when opening the browser and the connection not working

### DISCUSSION THREAD

**Discussion by WangJiaoJo** [verdaccio/verdaccio#3869] [Q&A] [answered]
Hi, Verdaccio team,

I have followed the instructions on this page to generate the certificates, 
<img width="884" alt="Screen Shot 2023-06-12 at 3 09 41 PM" src="https://github.com/verdaccio/verdaccio/assets/19795205/cda8ccb7-4450-44a2-94af-b7909ed510f2">
and when running Verdaccio, it is running on https://0.0.0.0:4873 but with warning. 
`warn --- http address - https://0.0.0.0:4873/ - verdaccio/5.25.0 `

And when visit this URL on browser, the page is like this and it cannot be connected only after clicking the 'Proceed to 0.0.0.0(unsafe)'. 
<img width="876" alt="Screen Shot 2023-06-12 at 2 44 01 PM" src="https://github.com/verdaccio/verdaccio/assets/19795205/5bc56e6f-5a81-417c-af70-f13fc6031cfb">
<img width="792" alt="Screen Shot 2023-06-12 at 2 44 06 PM" src="https://github.com/verdaccio/verdaccio/assets/19795205/9bc7ff53-548d-4f83-adf2-233f658e7b5c">

May I ask if there is any solution about this issue? Thanks in advance.

**Comment by juanpicado** [comment]
I frankly prefer using Docker example for https rather use this feature but I haven't tried since years so dunno so far.

**Reply by WangJiaoJo** [reply]
Hi, Juanpicado,
Thanks for your reply!
Is this an example that I can refer to for using htttps? 
https://github.com/verdaccio/verdaccio/blob/master/docker-examples/v4/https-portal-example/docker-compose.yml

Or could you please share the links if you have other examples that I can refer to?
I would really appreciae your help! Thanks!

**Reply by juanpicado** [reply, marked as answer]
Yeah that one, is a bit outdated but would work also on v5, also there are nginx examples you can also setup HTTPS at proxy level is the best way so far

**Reply by WangJiaoJo** [reply]
Thanks, Juanpicado, it is really helpful.

**Reply by WangJiaoJo** [reply]
@juanpicado Hi, I have a problem here, when I follow this example https://github.com/verdaccio/verdaccio/blob/master/docker-examples/v4/https-portal-example/docker-compose.yml, I get the error 
`cannot open config file /verdaccio/conf/config.yaml: false`
Here is my docker compose yml,
services:
   xxxxxx1:
     image:  appkit:latest
     restart: always
   verdaccio:
     depends_on:
       - xxxxxx1
     image: verdaccio:1.0.0
     container_name: xxxxxxxxx
     ports:
       - "4873:4873"
     restart: always
     volumes:
       - "./storage:/verdaccio/storage"
       - "./conf:/verdaccio/conf"
   https-portal:
     image: steveltn/https-portal:1
     ports:
       - '80:80'
       - '443:443'
     links:
       - verdaccio:verdaccio
     environment:
       DOMAINS: 'example.com -> http://verdaccio:4873'
       STAGE: local
       FORCE_RENEW: 'true'

Is there any wrong with the configuration or solution to resolve this error?
`'there are nginx examples you can also setup HTTPS at proxy level'`
And is it possible to share the example link or guidance of the nginx examples mentioned above?

Thanks in advance!

**Reply by juanpicado** [reply]
File permissions probably try chmod 777 to figure it out

**Reply by WangJiaoJo** [reply]
You mean use chips 777 to allow the permission to /verdaccio/conf files, right?
And any nginx examples that I can refer to?
Thanks!
